scipy/optimize/_linprog_util.py

```diff
@@ -729,19 +741,34 @@ def where(A):
         b_eq = b_eq - A_eq[:, i_f].dot(lb[i_f])
         b_ub = b_ub - A_ub[:, i_f].dot(lb[i_f])
         c = c[i_nf]
+        x_undo = lb[i_f]  # not x[i_f], x is just zeroes
         x = x[i_nf]
         # user guess x0 stays separate from presolve solution x
         if x0 is not None:
             x0 = x0[i_nf]
         A_eq = A_eq[:, i_nf]
         A_ub = A_ub[:, i_nf]
-        # record of variables to be added back in
-        undo = [np.nonzero(i_f)[0], lb[i_f]]
-        # don't remove these entries from bounds; they'll be used later.
-        # but we _also_ need a version of the bounds with these removed
+        # modify bounds
         lb_mod = lb[i_nf]
         ub_mod = ub[i_nf]
 
+        def rev(x_mod):
```

Comment:
Clearly I didn't know about `np.insert` when I wrote the original for loop that re-inserted variables. Also, I like `np.flatnonzero`. I've always used `np.where` but then I'm only ever interested in the 0th element of the result. This is much cleaner.